src/main/java/org/springframework/data/domain/Scroll.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.domain;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+import java.util.function.IntFunction;
+
+import org.springframework.data.util.Streamable;
+
+/**
+ * A scroll of data consumed from an underlying query result. A scroll is similar to {@link Slice} in the sense that it
+ * contains a subset of the actual query results for easier consumption of large result sets. The scroll is less
+ * opinionated about the actual data retrieval, whether the query has used index/offset, keyset-based pagination or
+ * cursor resume tokens.
+ *
+ * @author Mark Paluch
+ * @since 3.1
+ * @see ScrollPosition
+ */
+public interface Scroll<T> extends Streamable<T> {
```

Comment:
I think `Window` was the better therm here. It aligned nicely with `ScrollPosition`.